File: 프로그래머스/lv2/76502. 괄호 회전하기/괄호 회전하기.py
from collections import deque

def f(c):
    stack = []
    for j in c:
        if j in ['[','{','(']:
            stack.append(j)
        else:
            if len(stack) != 0:
                stack_pop = stack.pop()
                if j == ']' and stack_pop != '[':
                    return 0
                elif j == '}' and stack_pop != '{':
                    return 0
                elif j == ')' and stack_pop != '(':
                    return 0
            else:
                return 0
    if len(stack) == 0 :
        return 1
    else:
        return 0

def solution(s):
    ans = 0
    li = deque(list(s))
    for _ in range(len(s)):
        li.rotate(-1)
        ans += f(li)
    return ans

File: 프로그래머스/lv2/76502. 괄호 회전하기/괄호 회전하기.java
import java.util.Stack;

class Solution {
    public static boolean ifAnswer(char[] str){
        Stack<Character> stack = new Stack<>();

        for (char c: str){
            switch (c){
                case '[' -> stack.add(']');
                case '{' -> stack.add('}');
                case '(' -> stack.add(')');
                case ']', '}', ')' -> {
                    if (stack.isEmpty()) return false;
                    if (stack.pop() != c) return false;
                }
            }
        }
        return stack.isEmpty();
    }

    public static char[] CharToBack(char[] chars){
        if (chars.length <= 1) {
            return chars;
        }

        char firstChar = chars[0];

        for (int i = 0; i < chars.length - 1; i++) {
            chars[i] = chars[i + 1];
        }

        chars[chars.length - 1] = firstChar;
        return chars;
    }

    public static int solution(String s) {
        char[] str = s.toCharArray();
        int answer = 0;
        for (int i=0; i < s.length(); i++){
            if (ifAnswer(str)){
                answer += 1;
            }
            str = CharToBack(str);
        }

        return answer;
    }
}